packages/openapi-v3/package.json

```
@@ -26,6 +26,7 @@
     "@loopback/core": "^2.13.1"
   },
   "dependencies": {
+    "@loopback/repository": "^3.3.0",
```

Would Yarn2 PnP be satisfied with it being referenced as a `peerDependency` instead?

The reason for `peerDependency` is so that only a single version of the package is used across the project and its dependencies (See #5959). Adding `@loopback/repository` as a hard dependency would break this paradigm.

Though it got me curious if there's any other LB4 packages that have the same issue.

We also probably should update our CI pipeline to test with Yarn2 PnP.

```suggestion
    "@loopback/core": "^2.13.1"
    "@loopback/repository": "^3.3.0"
  },
  "dependencies": {
```

---

-1. Our extensions are intentionally using peer dependencies for `@loopback/repository`, as explained in the comment above.

To fix the problem you are observing with Yarn2 PnP, we should find the place that's depending on `@loopback/openapi-v3` but not fulfilling it's peer dependency on `@loopback/repository`. I think we need to add `@loopback/repository` to `peerDependencies` of `@loopback/rest`.

https://github.com/strongloop/loopback-next/blob/81fab94a16f09da8d6251cef68c33b127861c0d4/packages/rest/package.json#L26-L32

---

> We also probably should update our CI pipeline to test with Yarn2 PnP.

That's a great idea :+1: Would you @achrinza mind opening a new GH issue for that?